### ISSUE TITLE
Annotated pdf background not found if not residing in the same directory 

### ISSUE THREAD

**Issue by tjavdar** [xournalpp/xournalpp#3273]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Archlinux
 - (Linux only) i3 under X11
 - Which version of libgtk do you use:  3.24.30
 - Version of Xournal++: 1.1.0
 - Installation method: official Arch-repo package

**Describe the bug**

Annotating a pdf file does not save the full path of the pdf-background in the annotated xopp file.

**To Reproduce**
Goal: You wish to annotate ./abc.pdf and save the annotation in a subdir as xopp/abc.pdf.xopp.

1. xournal abc.pdf & 
2. # You annotate and save as:
    File->Save as->xopp/abc.pdf.xopp and close xournalpp
3. cd xopp/
4. xournalpp abc.pdf.xopp & 
5. # xournalpp throws an error: 
        "The background fiile abc.pdf could not be found, .... etc. It was last seen at: <empty>. Choose another PDF, etc.",
    # and creates a
       ~/.config/xournalpp/emergencysave.xopp

**Expected behavior**
   Annotated pdf-background should be saved with a full pathname in the annotating xopp-file, as in versions 1.0.*

**Additional context**
   Default filename suggestion for  an annotation of %.pdf should be %.pdf.xopp (and not %.xopp), as in versions 1.0.*

**Comment by rolandlo** [comment]
Looks like Xournal++ just uses the filename that is passed from the terminal. There are two workarounds:
1. Open the pdf with `File > Open` within Xournal++, instead of specifying the name in the terminal
2. When opening the file from the terminal, specify the full absolute filename of the PDF

**Comment by tjavdar** [comment]
Although this is a minor bug, seems it should be trivial to revert the behaviour of xournalpp 1.0.*, which did record the full filename.

Let me describe the workflow scenario when saving annotations in another directory seems to be desirable:

1. You produce ./abc.pdf from, e.g, a small LaTeX source, ./abc.tex
2. You annotate it and save the annotation as
             xopp/abc.xopp.pdf 
     and export the annotated file as
             xopp/abc.pdf
     which now does not filename-clash with the background.
3. You upload the annotation xopp/abc.pdf on platform your students follow
4.  You keep now only
         ./abc.tex  xopp/abc.pdf.xopp
     and  clean up everything  else (e.g. from a Makefile) because everything is reproducable 
          rm ./abc.{pdf,aux,dvi,log, any other LaTeX-garbage}
          rm xopp/abc.pdf

**Comment by rolandlo** [comment]
@tjavdar In your workflow it would be better to save the absolute path. There are other workflows, where it is the opposite, and probably that was the reason, why it was changed (one would have to look through related issues). When you save the xopp-file in the same folder as the pdf and you would like to be able to move them together to another folder, then the current behaviour is better.
Maybe a command line option for determining whether the pdf-filename should be relative or absolute would make sense.

**Comment by tjavdar** [comment]
I totally agree, probably keeping the pdf and the .xopp annotation in the same dir might very well be the most common (and desirable) scenario.

As for a solution for either scenario: A command line option, in my opinion, would be a a bloat and overkill.

Might it be better  to opt for a precedence solution, namely,  to save the absolute path of the annotated background, say

  ~/path/to/file.pdf

but use it only if neither

  ./background.xopp   -nor-  ./background.pdf.xopp

is present in the current directory. If present, either of these annotations takes precedence ofer the absolute path.
